### ISSUE TITLE
Possibility of creating a Kernel module?

### ISSUE THREAD

**Issue by JadinAndrews** [rdady/gear-vr-controller-linux#1]
Hi there, I have just stumbled on your project, I am honestly surprised that only about 3 people in the world (as far as I can tell) have tinkered with this controller. I think it would be great if this could be integrated as a kernel module, and used with the likes of retroarch for gaming. 

What do you think would be the challenges with this?

**Comment by rdady** [comment]
Indeed, I also would have counted on more interest, but it is always hard to predict what will gain popularity.
I don't think a kernel module for this device could be possible as it is Bluetooth based. If it was something like these wireless mouses like Rapoo with an own dongle capable of 2 addition Bluetooth based connections or a Jabra headset also with its own radio dongle plus a Bluetooth connection and a USB based connection. Well these can have their own kernel modules because of the (USB) dongles. But Gear VR has to deal with Bluetooth.
Although it is possiuble to build an intermediate device / dongle that acts as a joystick:
http://bbs.esp32.com/viewtopic.php?t=2162
Once this guy finishes USB software stack for ESP32 which is capable of Bluetooth like we need here, it does only take one cheap device from aliexpress some firmware upload and voila.

Alternatively:
Take ESP32 for Bluetooth and attach an Arduino Pro Micro (atmega32u4) for USB joystick emulation.
https://www.tinkerboy.xyz/how-to-program-the-pro-micro-atmega32u4-as-a-usb-gamepad-controller-with-arduino/
As the ESP32 has a WIFI port, it could have a web server running the Gear VR - joystick mapping you wish for the moment.
